api/src/main/java/jakarta/servlet/ServletConnection.java

```diff
@@ -0,0 +1,88 @@
+     * being used for the {@code ServletRequest} from which this {@code ServletConnection} was obtained.
+     * <p>
+     * There is no defined format for this string. The format is implementation dependent.
+     * 
+     * @return A unique identifier for the network connection
+     */
+    String getConnectionId();
+
+    /**
+     * Obtain a unique (within the lifetime of the JVM) identifier string for the {@code ServletRequest} from which this
+     * {@code ServletConnection} was obtained.
+     * <p>
+     * There is no defined format for this string. The format is implementation dependent.
+     * 
+     * @return A unique identifier for the servlet request
+     */
+    String getRequestId();
+
+    /**
+     * Obtain the name of the protocol as presented to the JVM after the removal, if present, of any TLS encryption. This
+     * may not be the same as the protocol seen by the application. For example, a reverse proxy may present AJP whereas the
+     * application will see HTTP 1.1.
+     * <p>
+     * If the protocol has an entry in the <a href=
+     * "https://www.iana.org/assignments/tls-extensiontype-values/tls-extensiontype-values.xhtml#alpn-protocol-ids">IANA
+     * registry for ALPN names then the identification sequence, in string form, must be returned. Registered identification
+     * sequences MUST only be used for the associated protocol. Return values for other protocols are implementation
+     * dependent. Unknown protocols should return the string "unknown".
+     * 
+     * @return The name of the protocol presented to the JVM after decryption of TLS, if any.
+     */
+    String getProtocol();
+
+    /**
+     * Obtain the connection identifier for the network connection to the JVM that is being used for the
+     * {@code ServletRequest} from which this {@code ServletConnection} was obtained as defined by the protocol in use. Note
+     * that some protocols do not define such an identifier.
+     * 
+     * @return The connection identifier if one is defined, otherwise an empty string
+     */
+    String getProtocolConnectionId();
+
+    /**
+     * Obtain the request identifier for the {@code ServletRequest} from which this {@code ServletConnection} was obtained
+     * as defined by the protocol in use. Note that some protocols do not define such an identifier.
+     * 
+     * @return The request identifier if one is defined, otherwise an empty string
+     */
+    String getProtocolRequestId();
+
+    /**
+     * Determine whether or not the incoming network connection to the JVM used TLS or not. Note that where a reverse proxy
+     * is used, the application may have a different view as to whether TLS is being used due to the use of headers like
+     * {@code X-Forwarded-Proto}
+     * 
+     * @return {@code true} if the incoming network connection used TLS, otherwise {@code false}
+     */
+    boolean getTLS();
```

Comment:
is `TLS` the right name for this?   SSL was replaced by TLS and TLS might eventually be replaced by something else.   I think h3 on quic is still considered TLS, but it loosens the dependency.

How about `isSecure()` or `isProtocolSecure()`?

Reply:
I prefer the former as the latter creates the impression of a link with `getProtocol()` and I was trying to keep them independent.